src/sentry/api/endpoints/team_details.py

```diff
@@ -56,6 +56,11 @@ class TeamDetailsEndpoint(TeamEndpoint):
         "GET": ApiPublishStatus.PUBLIC,
         "PUT": ApiPublishStatus.PUBLIC,
     }
+    # OrganizationSCIMTeamDetails inherits this endpoint, but toggles this setting
+    _allow_idp_changes = False
+
+    def is_permitted_idp_change(self, team: Team):
```

Comment:
i got confused by what this does, but i can't think of a great name for this (`can_modify_idp_team`?). at least the function contents are straightforward

Reply:
Changed it to `can_modify_team` and added an early exit for non-idp teams since the boolean logic was flawed before.